src/key_value_db.rs

```diff
@@ -240,6 +244,9 @@ where
         self.snap_counter += 1;
     }
 
+    /// Retrieves a transactional state using the closest snapshot.
```

Comment:
Worth precising that if the snapshot isn't in the exact id we take changes to go back to a special id

Reply:
Updated the comments to reflect that :+1: